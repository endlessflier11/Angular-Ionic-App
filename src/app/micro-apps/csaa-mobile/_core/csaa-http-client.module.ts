import { NgModule } from '@angular/core';
import { AngularHttpAdapter } from './services/http/adapters/angular-http.adapter';
import { HttpClient, HttpClientModule } from '@angular/common/http';
import { IonicHttpAdapter } from './services/http/adapters/ionic-http.adapter';
import { HTTP } from '@ionic-native/http/ngx';
import { UNI_HTTP_ADAPTER } from './services/http/uni-http.model';

@NgModule({
  imports: [HttpClientModule],
  providers: [
    HTTP,
    { provide: UNI_HTTP_ADAPTER, multi: true, useClass: AngularHttpAdapter, deps: [HttpClient] },
    {
      provide: UNI_HTTP_ADAPTER,
      multi: true,
      useClass: IonicHttpAdapter,
      deps: [HTTP],
    },
  ],
  exports: [HttpClientModule],
})
export class CsaaHttpClientModule {}
