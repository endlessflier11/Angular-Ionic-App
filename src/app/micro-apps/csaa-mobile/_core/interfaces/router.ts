import { Route, UrlTree } from '@angular/router';
import { NavigationOptions } from '@ionic/angular/providers/nav-controller';
import { Observable } from 'rxjs';

interface AppRoute extends Route {
  name?: string;
  isGroupIndex?: boolean; // If true, it means route is the index of parent. Then it inherits all parent properties
}

export type AppRoutes = AppRoute[];

export interface RouteParam {
  [key: string]: string;
}

export interface NamedRouteConfig {
  name: string;
  path: string;
  children?: NamedRouteConfig[];
}

export interface NamedRoute {
  name: string;
  path: string;
  relativePath: string;
}

export interface CanComponentDeactivate {
  canDeactivate: () =>
    | Observable<boolean | UrlTree>
    | Promise<boolean | UrlTree>
    | boolean
    | UrlTree;
}

export interface CsaaRouteStackItem {
  name: string;
  params?: RouteParam;
  options?: NavigationOptions;
}
