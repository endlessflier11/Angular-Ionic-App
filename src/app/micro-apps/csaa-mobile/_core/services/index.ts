export * from './router/router.service';
export * from './config/config.service';
export * from './auth/auth.service';
export * from './sso/sso.service';
export * from './storage/storage.service';
export * from './http/http.service';
export * from './analytics.service';
export * from './policy.service';
export * from './payment.service';
export * from './global-state.service';
export * from './call.service';
export * from './metadata.service';
export * from './pdf-display.service';
export * from './webview/webview.service';
