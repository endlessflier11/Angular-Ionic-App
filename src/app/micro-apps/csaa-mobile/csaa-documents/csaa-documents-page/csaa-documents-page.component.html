<csaa-loading [visible]="refreshingUri$ | async"></csaa-loading>
<ion-header [ngClass]="['csaa', currentTheme]">
  <ion-toolbar>
    <ion-buttons slot="start">
      <csaa-back-button (back)="backButtonClick()"></csaa-back-button>
    </ion-buttons>
  </ion-toolbar>
  <ion-toolbar>
    <ion-title>Documents</ion-title>
  </ion-toolbar>
</ion-header>
<ion-content [ngClass]="['csaa', currentTheme]">
  <csaa-skeleton-card *ngIf="loading"></csaa-skeleton-card>
  <ion-refresher (ionRefresh)="documentsRefresh($event)" slot="fixed">
    <ion-refresher-content></ion-refresher-content>
  </ion-refresher>
  <ion-card class="csaa-card coverage-list" *ngIf="!loading">
    <ion-card-content class="csaa-item-list">
      <ion-list>
        <ion-item *ngIf="policy$ | async as policy" lines="none">
          <ion-label>
            <h3>{{ policy.number }}</h3>
            <span class="subtitle">
              <csaa-policy-subtitles [policySubtitle]="policy.subtitle"></csaa-policy-subtitles>
            </span>
          </ion-label>
          <csaa-theme-icon slot="end" [type]="policy.type" [size]="40"> </csaa-theme-icon>
        </ion-item>

        <ion-item detail class="filter-by">
          <ion-label>
            <ion-text> Filter by </ion-text>
          </ion-label>
          <span (click)="openModal()">{{ getFilterLabel(activeFilter$ | async) }}</span>
        </ion-item>
        <ion-item lines="none" class="filter-by-current">
          <ion-label [ngClass]="{ 'view-archive': (viewArchive$ | async) }"
            ><ion-text>Current</ion-text>
            <ion-toggle (ionChange)="viewArchiveChanged($event)"></ion-toggle>
            <ion-text>Archive</ion-text></ion-label
          >
        </ion-item>

        <ion-item
          detail
          *ngFor="let document of limitedDocuments$ | async"
          (click)="clickDocumentsForPdf(document)"
        >
          <ion-label>
            <ion-text class="document-title" color="primary">
              <span class="title">{{ document.docName }}</span>
            </ion-text>

            <span class="subtitle">
              Effective Date {{ document.agreementEffectiveDate | date: 'MM/dd/yyyy' }}
            </span>
            <span class="subtitle">
              Process Date {{ document.processDate | date: 'MM/dd/yyyy' }}
            </span>
          </ion-label>
          <span slot="end" class="document-status" *ngIf="document.isNew">New</span>
        </ion-item>
        <ion-button
          expand="full"
          fill="clear"
          (click)="showMore()"
          *ngIf="(currentLimit$ | async) < (documents$ | async)?.length"
          >MORE</ion-button
        >
      </ion-list>
    </ion-card-content>
  </ion-card>
</ion-content>
