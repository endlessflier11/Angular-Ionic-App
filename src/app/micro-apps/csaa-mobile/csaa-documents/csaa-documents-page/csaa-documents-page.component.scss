@import '../../_scss/variables';

ion-label {
  white-space: normal !important;
}

ion-text.document-title {
  display: flex;
  flex-flow: row;
  align-items: center;
  line-height: 28px;

  .title {
    flex-grow: 1;
    color: var(--ion-color-primary, $csaa-active-color);
    font-weight: $csaa-bolder-font-weight;
    font-size: 16px;
  }
}

span.subtitle {
  display: block;
  text-decoration: none;
  line-height: 22px;
}

.filter-by, .filter-by-current {
  ion-text {
    font-size: 16px;
  }
  span {
    text-decoration: none;
    background-color: var(--ion-color-primary-contrast);
    font-size: 16px;
  }
}

.filter-by-current {
  ion-label {
    display: flex;
    justify-content: space-between;
    align-items: center;

    &:not(.view-archive) {
      ion-text:last-child {
        color: $csaa-lighter-color;
      }
    }

    &.view-archive {
      ion-text:first-child {
        color: $csaa-lighter-color;
      }
    }
  }
}
