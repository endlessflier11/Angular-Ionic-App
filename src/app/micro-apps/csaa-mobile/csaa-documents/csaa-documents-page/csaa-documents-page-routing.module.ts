import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';
import { CsaaDocumentsPageComponent } from './csaa-documents-page.component';

const routes: Routes = [
  {
    path: '',
    component: CsaaDocumentsPageComponent,
  },
];

@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule],
})
export class CsaaDocumentsPageRoutingModule {}
