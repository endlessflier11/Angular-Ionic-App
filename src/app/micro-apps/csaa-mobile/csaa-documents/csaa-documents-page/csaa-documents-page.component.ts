import { Component, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { BehaviorSubject, combineLatest, noop, Observable, Subscription } from 'rxjs';
import { unsubscribeIfPresent, withErrorReporter } from '../../_core/helpers';
import { AppEndpointsEnum, Category, EventName, EventType, Policy } from '../../_core/interfaces';
import { PolicyDocument, PolicyDocumentType } from '../../_core/interfaces/document.interface';
import {
  AnalyticsService,
  ConfigService,
  HttpService,
  PdfDisplayService,
  RouterService,
} from '../../_core/services';
import { Select, Store } from '@ngxs/store';
import { PolicyState } from '../../_core/store/states/policy.state';
import { CustomerAction, PolicyAction } from '../../_core/store/actions';
import { finalize, map, mergeMap, switchMap } from 'rxjs/operators';
import { FetchState } from '../../_core/store/states/fetch.state';
import { interactWithLoader } from '../../_core/operators';
import { IonRefresher, ModalController, Platform } from '@ionic/angular';
import { DocumentModalComponent } from '../document-modal/document-modal.component';
import { DOCUMENT_ACTION_MODAL_ID } from '../../_core/ui-kits/document-action-modal/document-action-modal.component';
import { ConfigSelector } from '../../_core/store/selectors';

const ITEM_LIMIT = 50;

@Component({
  selector: 'csaa-csaa-documents-page',
  templateUrl: './csaa-documents-page.component.html',
  styleUrls: ['./csaa-documents-page.component.scss'],
})
export class CsaaDocumentsPageComponent implements OnInit {
  @Select(FetchState.isLoading(PolicyAction.RefreshActiveDocument))
  refreshingUri$: Observable<boolean>;
  policy$: Observable<Policy>;
  documents$: Observable<PolicyDocument[]>;
  limitedDocuments$: Observable<PolicyDocument[]>;
  activeFilter$: BehaviorSubject<string> = new BehaviorSubject('all_documents');
  filterBy: any;

  currentTheme: string;
  policyNumber: string;
  currentFilter: string;
  loading = false;
  backButtonSubscription: Subscription;
  currentLimit$ = new BehaviorSubject(ITEM_LIMIT);
  viewArchive$ = new BehaviorSubject(false);

  public toggleControlValue = false;
  private refresher: IonRefresher;

  constructor(
    private analyticsService: AnalyticsService,
    private configService: ConfigService,
    private routerService: RouterService,
    private route: ActivatedRoute,
    private store: Store,
    private platform: Platform,
    private pdfDisplayService: PdfDisplayService,
    private modalCtrl: ModalController
  ) {
    this.loading = true;
    this.currentTheme = this.configService.getTheme();
  }
  ionViewWillEnter() {
    this.backButtonSubscription = this.platform.backButton.subscribeWithPriority(10, () =>
      this.backButtonClick()
    );
  }

  ionViewWillLeave() {
    unsubscribeIfPresent(this.backButtonSubscription);
  }

  ngOnInit() {
    this.route.paramMap.subscribe(
      withErrorReporter((paramMap) => {
        this.policyNumber = paramMap.get('policyNumber');
        this.policy$ = this.store.select(PolicyState.policyData(this.policyNumber));
        this.documents$ = combineLatest([
          this.store.select(PolicyState.documentsForPolicy(this.policyNumber)),
          this.activeFilter$,
          this.viewArchive$,
        ]).pipe(
          map(([documents, filterBy, viewArchive]) =>
            this.applyActiveFilter(documents, filterBy, viewArchive)
          )
        );
        this.limitedDocuments$ = combineLatest([this.documents$, this.currentLimit$]).pipe(
          map(([documents, currentLimit]) => documents.slice(0, currentLimit))
        );
        this.dispatchLoadAction();
      })
    );
  }

  documentsRefresh(event) {
    this.loading = true;
    this.refresher = event.target;
    this.dispatchReloadAction();
  }

  applyActiveFilter(documents: PolicyDocument[], filterBy: string, viewArchive: boolean) {
    const currentDocuments = documents.filter(
      (d) => (d.current && !viewArchive) || (!d.current && viewArchive)
    );

    if (filterBy === 'all_documents') {
      return currentDocuments;
    } else if (filterBy === 'billing_&_payments') {
      return currentDocuments.filter((d) => d.category === 'Billing & Payments');
    } else if (filterBy === 'policy_documents') {
      return currentDocuments.filter((d) => d.category === 'Policy Documents');
    }
    return currentDocuments.filter((d) => d.docType === filterBy);
  }

  getFilterLabel(filterBy: string) {
    if (filterBy === 'all_documents') {
      return 'All Documents';
    } else if (filterBy === 'billing_&_payments') {
      return 'Billing & Payments';
    } else if (filterBy === 'policy_documents') {
      return 'Policy Documents';
    }
    return PolicyDocumentType[filterBy];
  }

  clickDocumentsForPdf(policyDocument: PolicyDocument) {
    const policy = this.store.selectSnapshot(PolicyState.policyData(this.policyNumber));
    this.pdfDisplayService
      .showDocumentActionPrompt(policyDocument.docName)
      .subscribe(({ selection, fileName }) => {
        this.store
          .dispatch(new PolicyAction.RefreshActiveDocument(this.policyNumber, policyDocument))
          .pipe(
            mergeMap(() => {
              const externalURI = this.store.selectSnapshot(PolicyState.activeDocumentUri);
              const {
                appEndpoints: { endpoints },
              } = this.store.selectSnapshot(ConfigSelector.state);

              // We can add a property to suggest whether user choose to save only or to view in addition
              this.analyticsService.trackEvent(EventName.DOCUMENT_SELECTED, Category.documents, {
                event_type: EventType.OPTION_SELECTED,
                selection: policyDocument.docName,
                document_category: policyDocument.category,
                document_type: PolicyDocumentType[policyDocument.docType],
                document_name: policyDocument.docName,
                document_effective: policyDocument.current ? 'current' : 'archive',
                process_date: policyDocument.processDate,
                ...AnalyticsService.mapPolicy(policy),
              });

              const url = HttpService.compileUrlWithParams(
                endpoints[AppEndpointsEnum[AppEndpointsEnum.policyDocumentBaseUrl]],
                { externalURI }
              );
              return this.pdfDisplayService
                .accessPdfDocument(url, selection, fileName)
                .pipe(interactWithLoader());
            })
          )
          .subscribe(
            withErrorReporter(
              (result) => {
                this.pdfDisplayService.onActionSuccess(policy, policyDocument, selection, result);
              },
              (error) => {
                this.pdfDisplayService.onActionFailed(policy, policyDocument, error);
              }
            )
          );
      });
  }

  backButtonClick() {
    this.analyticsService.trackEvent(EventName.HOME_ACCESSED, Category.global, {
      event_type: EventType.LINK_ACCESSED,
      link: 'Home',
    });
    this.modalCtrl.dismiss(undefined, undefined, DOCUMENT_ACTION_MODAL_ID).catch(noop).then(noop);
    this.routerService.back();
  }

  private dispatchLoadAction() {
    this.loading = true;
    this.store
      .dispatch(new CustomerAction.LoadCustomer())
      .pipe(
        switchMap(() => this.store.dispatch(new PolicyAction.LoadPolicies())),
        switchMap(() =>
          this.store.dispatch(new PolicyAction.LoadPolicyDocuments(this.policyNumber))
        ),
        finalize(() => this.completeLoading())
      )
      .subscribe(withErrorReporter());
  }

  private dispatchReloadAction() {
    this.loading = true;
    this.store
      .dispatch(new CustomerAction.LoadCustomer())
      .pipe(
        switchMap(() => this.store.dispatch(new PolicyAction.LoadPolicies())),
        switchMap(() =>
          this.store.dispatch(new PolicyAction.ReloadPolicyDocuments(this.policyNumber))
        ),
        finalize(() => this.completeLoading())
      )
      .subscribe(withErrorReporter());
  }

  private completeLoading() {
    this.loading = false;
    if (this.refresher) {
      this.refresher.complete().then(noop);
    }
  }

  async openModal() {
    const policy = this.store.selectSnapshot(PolicyState.policyData(this.policyNumber));
    const docTypes = new Set(
      this.store
        .selectSnapshot(PolicyState.documentsForPolicy(this.policyNumber))
        .filter((d) => (this.viewArchive$.getValue() ? !d.current : d.current))
        .map((d) => d.docType)
    );

    const modal = await this.modalCtrl.create({
      component: DocumentModalComponent,
      componentProps: {
        currentFilter: Array.from(docTypes),
        policyType: policy.type,
        activeFilter: this.activeFilter$.getValue(),
      },
    });

    modal.onWillDismiss().then((docTypeReturned) => {
      if (docTypeReturned?.data) {
        this.activeFilter$.next(docTypeReturned.data);
        this.currentLimit$.next(ITEM_LIMIT);
      }
    });

    await modal.present();
  }

  showMore() {
    this.currentLimit$.next(this.currentLimit$.getValue() + ITEM_LIMIT);
  }

  viewArchiveChanged(event) {
    this.viewArchive$.next(event?.detail?.checked);
    this.currentLimit$.next(ITEM_LIMIT);
  }
}
