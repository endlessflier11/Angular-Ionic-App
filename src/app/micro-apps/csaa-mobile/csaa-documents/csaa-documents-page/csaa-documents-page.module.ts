import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { IonicModule } from '@ionic/angular';
import { CsaaDocumentsPageComponent } from './csaa-documents-page.component';
import { CsaaDocumentsPageRoutingModule } from './csaa-documents-page-routing.module';
import { UiKitsModule } from '../../_core/ui-kits/ui-kits.module';
import { DocumentModalComponent } from '../document-modal/document-modal.component';

@NgModule({
  declarations: [CsaaDocumentsPageComponent, DocumentModalComponent],
  imports: [
    CommonModule,
    CsaaDocumentsPageRoutingModule,
    FormsModule,
    IonicModule,
    UiKitsModule,
    // CollapsedCardModule,
    // CardListModule,
    // SpinnerModule,
    // PolicyStatusCardModule,
  ],
})
export class CsaaDocumentsPageModule {}
