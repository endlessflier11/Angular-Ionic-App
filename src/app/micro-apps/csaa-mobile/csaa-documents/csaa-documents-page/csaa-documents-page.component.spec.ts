/* eslint-disable max-len */
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { IonicModule } from '@ionic/angular';

import { CsaaDocumentsPageComponent } from './csaa-documents-page.component';
import { UiKitsModule } from '../../_core/ui-kits/ui-kits.module';
import { PdfDisplayService, PolicyService } from '../../_core/services';
import { ActivatedRoute } from '@angular/router';
import { Store } from '@ngxs/store';
import { PageTestingModule, StoreTestBuilder } from '@app/testing';
import { of } from 'rxjs';
import { CsaaAppInjector } from '../../csaa-app.injector';
import { Injector } from '@angular/core';

describe('CsaaDocumentsPageComponent', () => {
  let component: CsaaDocumentsPageComponent;
  let fixture: ComponentFixture<CsaaDocumentsPageComponent>;
  let store: Store;
  const PARAM_MAP = new Map();

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [CsaaDocumentsPageComponent],
      imports: [
        IonicModule,
        UiKitsModule,
        PageTestingModule.withConfig({
          providesStorage: true,
          providesAnalytics: true,
          providesConfig: true,
          providesRouter: true,
        }),
      ],
      providers: [
        {
          provide: ActivatedRoute,
          useValue: {
            paramMap: of(PARAM_MAP),
          },
        },
        { provide: PdfDisplayService, useValue: { viewPdf: jest.fn() } },
        {
          provide: PolicyService,
          useValue: { getDocumentsForPolicy: jest.fn().mockReturnValue(of([])) },
        },
      ],
    }).compileComponents();

    store = TestBed.inject(Store);
    StoreTestBuilder.withDefaultMocks().resetStateOf(store);
    CsaaAppInjector.injector = TestBed.inject(Injector);

    PARAM_MAP.set('policyNumber', 'CAAC910026006');
    PARAM_MAP.set('policyType', '2');

    fixture = TestBed.createComponent(CsaaDocumentsPageComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should load documents data', () => {
    // TODO: FIX THIS DOCUMENTS SELECTION FROM STORE
    component.documents$ = of([
      {
        agreementEffectiveDate: '2020-07-21',
        category: 'Policy Documents',
        contentType: 'application/pdf',
        current: true,
        description: 'Property Declaration Page',
        docName: 'Property Declaration Page',
        docType: 'DeclarationsPage',
        externalURI:
          'policy/documentsecure/AAAADHIsGr6kQ-NZzYA3lEZqh0lk-qegJO8E9weR4ZvWAWwRYYXN2tOelxBjBEzaGkRdnB6govDWcDtGn1Y3javHNruvTFuD2nNB7Wg2eB5CYCv1SIrxAg/2021-06-01T14:06:42.691154-07:00/bd29bd470e54c9578321fc464594b075f3b38e0e2924d8e5113c061a317dd1e5',
        formId: 'HS02_4',
        isNew: false,
        oid: '6fa2a3fe3f96875948e9033ab109c5c91a70adbf35828d85fb7f379b98b71a61',
        policyNumber: 'WYH4910013941',
        processDate: '2020-09-08',
      },
      {
        agreementEffectiveDate: '2020-07-21',
        category: 'Policy Documents',
        contentType: 'application/pdf',
        current: true,
        description: 'Property Declaration Page',
        docName: 'Property Declaration Page',
        docType: 'DeclarationsPage',
        externalURI:
          'policy/documentsecure/AAAADHPLdLr3BP43aXcBusxPpW-GaGZBIawINmmzRMhvu0if2aPyq3jmShacNT07hNxHzN9uu6ydcc0iINH6yoi9yFlkSm-PkfaPn0UxWjreQeSxzrv2Vw/2021-06-01T14:06:42.692028-07:00/cc860e52960ac340931a535eeba2143e5603d9faa3b209a65bd021d93e7de39c',
        formId: 'HS02_4',
        isNew: false,
        oid: 'bf00c590069554cb09d67925577a3aa25eda3831ba948a09a9cc252d49502371',
        policyNumber: 'WYH4910013941',
        processDate: '2020-09-08',
      },
    ]);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
