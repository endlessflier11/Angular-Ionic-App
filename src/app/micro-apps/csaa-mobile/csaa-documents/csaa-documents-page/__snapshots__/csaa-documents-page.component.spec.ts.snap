// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CsaaDocumentsPageComponent should load documents data 1`] = `
<csaa-csaa-documents-page
  __ngContext__={[Function LRootView]}
  __refreshingUri$__selector={[Function Observable]}
  activeFilter$={[Function BehaviorSubject]}
  analyticsService={[Function AnalyticsMockService]}
  configService={[Function ConfigMockService]}
  currentLimit$={[Function BehaviorSubject]}
  currentTheme=""
  documents$={[Function Observable]}
  limitedDocuments$={[Function Observable]}
  loading="false"
  modalCtrl={[Function ModalController]}
  pdfDisplayService={[Function Object]}
  platform={[Function Platform]}
  policy$={[Function Observable]}
  policyNumber={[Function String]}
  route={[Function Object]}
  routerService={[Function RouterMockService]}
  store={[Function Store]}
  toggleControlValue="false"
  viewArchive$={[Function BehaviorSubject]}
>
  <csaa-loading>
  </csaa-loading><ion-header
    class="csaa"
  >
    <ion-toolbar>
      <ion-buttons
        slot="start"
      >
        <csaa-back-button>
          <ion-button>
            <ion-icon
              name="chevron-back-outline"
              slot="start"
            />
             Back
          </ion-button>
        </csaa-back-button>
      </ion-buttons>
    </ion-toolbar>
    <ion-toolbar>
      <ion-title>
        Documents
      </ion-title>
    </ion-toolbar>
  </ion-header><ion-content
    class="csaa"
  >
    <ion-refresher
      slot="fixed"
    >
      <ion-refresher-content />
    </ion-refresher>
    <ion-card
      class="csaa-card coverage-list"
    >
      <ion-card-content
        class="csaa-item-list"
      >
        <ion-list>
          <ion-item
            lines="none"
          >
            <ion-label>
              <h3>
                CAAC910026006
              </h3>
              <span
                class="subtitle"
              >
                <csaa-policy-subtitles>
                  <span
                    class="subtitle"
                  >
                    2009 Acura Tsx,
                  </span>
                  <span
                    class="subtitle"
                  >
                    2007 Jeep Grand Cherokee
                  </span>
                  <a
                    class="csaa-link"
                    href="#"
                  >
                    <small>
                      Show more
                    </small>
                  </a>
                </csaa-policy-subtitles>
              </span>
            </ion-label>
            <csaa-theme-icon
              slot="end"
            >
              <img
                src="/assets/csaa-mobile/vectors/icn-Auto.svg"
                style="width: 40px; height: 40px;"
              />
            </csaa-theme-icon>
          </ion-item>
          <ion-item
            class="filter-by"
            detail=""
          >
            <ion-label>
              <ion-text>
                 Filter by 
              </ion-text>
            </ion-label>
            <span>
              All Documents
            </span>
          </ion-item>
          <ion-item
            class="filter-by-current"
            lines="none"
          >
            <ion-label>
              <ion-text>
                Current
              </ion-text>
              <ion-toggle />
              <ion-text>
                Archive
              </ion-text>
            </ion-label>
          </ion-item>
        </ion-list>
      </ion-card-content>
    </ion-card>
  </ion-content>
</csaa-csaa-documents-page>
`;
